bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -164,52 +184,66 @@ pub mod pallet {
 			let asset_location: Location =
 				(*asset_id).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
 
-			let mut checked = false;
-			if asset_location.eq(&origin_location) || asset_location.starts_with(&origin_location) {
-				checked = true
-			}
-			ensure!(checked, <Error<T>>::OwnerCheck);
+			// Burn Ether Fee for the cost on ethereum
+			Self::burn_for_teleport(&origin_location, &(T::EthereumLocation::get(), fee).into())?;
 
-			Self::burn_fees(origin_location.clone(), fee)?;
+			let reanchored_asset_location = Self::reanchor(&asset_location)?;
 
-			let call = SnowbridgeControl::Control(ControlCall::RegisterToken {
-				asset_id: Box::new(VersionedLocation::from(asset_location.clone())),
+			let call = BridgeHubRuntime::EthereumSystem(EthereumSystemCall::RegisterToken {
+				asset_id: Box::new(VersionedLocation::from(reanchored_asset_location.clone())),
 				metadata,
 				fee,
 			});
 
-			let xcm: Xcm<()> = vec![
-				ReceiveTeleportedAsset(T::DeliveryFee::get().into()),
-				PayFees { asset: T::DeliveryFee::get() },
-				Transact {
-					origin_kind: OriginKind::Xcm,
-					call: call.encode().into(),
-					fallback_max_weight: None,
-				},
-			]
-			.into();
+			let message_id = Self::send(origin_location.clone(), Self::build_xcm(call.encode()))?;
 
-			Self::send(xcm)?;
-
-			Self::deposit_event(Event::<T>::RegisterToken { location: asset_location });
+			Self::deposit_event(Event::<T>::RegisterToken { location: asset_location, message_id });
 
 			Ok(())
 		}
 	}
 
 	impl<T: Config> Pallet<T> {
-		pub fn send(xcm: Xcm<()>) -> DispatchResult {
-			let bridgehub = Location::new(1, [Parachain(1002)]);
-			send_xcm::<T::XcmSender>(bridgehub, xcm).map_err(|_| Error::<T>::Send)?;
-			Ok(())
+		fn send(origin: Location, xcm: Xcm<()>) -> Result<H256, Error<T>> {
+			let (message_id, price) =
+				send_xcm::<T::XcmSender>(T::BridgeHubLocation::get(), xcm.clone()).map_err(
+					|err| {
+						tracing::error!(target: LOG_TARGET, ?err, ?xcm, "XCM send failed with error");
+						Error::<T>::from(err)
+					},
+				)?;
+			T::XcmExecutor::charge_fees(origin, price).map_err(|_| Error::<T>::FundsUnavailable)?;
```

Comment:
Should mirror inbound-queue:

```suggestion
			T::XcmExecutor::charge_fees(origin, price).map_err(|_| Error::<T>::FeesNotMet)?;
```

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/00f5ac8081c229591ba3dcb3a85795e343c37904